ngrinder-core/src/main/java/net/grinder/engine/agent/FileStore.java

```diff
@@ -144,7 +144,13 @@ public void handle(RefreshCacheMessage message) {
 						List<File> cachedFiles = getAllFilesInDirectory(cacheDir);
 						cachedFiles
 							.stream()
-							.filter(file -> !isRequiredFile(requiredFilesDigest, file))
+							.filter(file -> {
+								try {
+									return !requiredFilesDigest.contains(getFileDigest(cacheDir, file));
+								} catch (IOException e) {
+									throw new NGrinderRuntimeException(e);
+								}
+							})

```

Comment:
It would be better to change `isRequiredFile()` method to simplify lambda expr. Or remove the `isRequireFile()` method that is unused.